### ISSUE TITLE
Orthogonal viewer plugin (simultaneous orthogonal views)

### ISSUE THREAD

**Issue by grlee77** [napari/napari#1478]
I first want to congratulate the napari devs for their great work on this package. It is very impressive!

I work in medical imaging and often collaborate with @jni on scikit-image, but for some reason had not gotten around to trying napari. I tried it out yesterday on some large (~16GB) 3D data and browsing through slices was very smooth! Sliding and zooming the volume worked well. I think the only mouse-control I am used to having in other applications that I didn't see was the ability to adjust contrast and window level by dragging the mouse in combination with one of the mouse buttons (I did see there is a slider for it in the left panel).

As you may be aware, there are quite a few good standalone viewers/tools developed in C++ for medical imaging applications (e.g. [freeview](https://surfer.nmr.mgh.harvard.edu/fswiki/FreeviewGuide), [FSLeyes](https://users.fmrib.ox.ac.uk/~paulmc/fsleyes/userdoc/latest/overview.html), [3D Slicer](https://www.slicer.org/), [Medinria](https://med.inria.fr/), etc.), but I am excited to see something in the Python space. It is also great that this viewer can be called directly on general NumPy and Dask arrays as opposed to requiring specific medical imaging file formats.

## 🚀 Feature

I found the following duplicate issue in an archived repository: https://github.com/napari/napari-core/issues/4,

Do you know if there is anyone working on a plugin that would show simultaneous orthogonal views of 3D data (e.g. similar to [this screenshot from 3D slicer](https://www.slicer.org/w/img_auth.php/2/2c/20180426_DataLoadingAndVisualizationTutorial.png), although as a first step it could exclude the 4th panel with the 3D rendering)? Another option sometimes used for a 4th panel is to show the time-course of the selected 3D spatial position when the data is 3D + time.

Perhaps this is already possible today? I have only just started looking at napari. I saw that you can already switch between the orthogonal views in the main window, but it would be nice to be able to see them side by side and have the spatial positions linked across the views.

## Motivation

[nibabel](https://nipy.org/nibabel/reference/nibabel.viewers.html) provides a basic 3D orthogonal viewer built using Matplotlib, but Napari could potentially produce something much nicer.

## Pitch

A key feature is that clicking on a specific voxel in one view, updates the other two orthogonal views to show the slice at the selected location. Often there is an (optional) [crosshair overlaid on each view](https://users.fmrib.ox.ac.uk/~paulmc/fsleyes/userdoc/latest/_images/overview_ortho.png) to show the current location across views. 

## Alternatives

I think the main argument against developing this is that there are already good non-Python tools that do this. I think there is still an argument that Napari could provide similar functionality, but in a less domain-specific manner and with the flexibility to combine interactively with other Python image processing tools.

**Comment by grlee77** [comment]
I don't foresee having time to work on this personally in the immediate future, but if there is interest from the team here, it could be worth advertising Napari on the [Neuroimaging Analysis in Python mailing list](https://mail.python.org/mailman/listinfo/neuroimaging) to help find new users and potential contributors.

**Comment by tlambert03** [comment]
Thanks @grlee77!  Glad you've been finding it useful.  Orthoviews are definitely a very high priority for us (see also #760 and #662) and agree that tons of people could use that functionality.   It probably could be cobbled together with the current codebase, but we have a few deeper architectural things we need to shore up (e.g. #1276) that would make it much more natural.  Thanks for bumping the issue and providing the useful context.

**Comment by sofroniewn** [comment]
Yes, as @tlambert03 said, it's definitely on our priority list, but we'll probably in a better position after things like #1353 merge too and we clean up our slicing model. I'll be very excited when we having this functionality too.

> if there is interest from the team here, it could be worth advertising Napari on the Neuroimaging Analysis in Python mailing list to help find new users and potential contributors.

Thanks, that sounds like a great idea, we're always interested in new user and contributors and there are certainly some neuroimaging specific file IO plugins that community might be interested in working on right now as we prepare for ortho views!!

**Comment by tibuch** [comment]
Hello everyone,

An orthogonal viewer would be very useful for some of my users and I wanted to inquire what the status/timeline is of this feature? Are there still some larger underlying issues to be resolved first or would it be possible to start working on this already? I would also be able to contribute some developer time to get this done if necessary. However if the underlying code is undergoing heavy refactoring I would not start yet.

**Comment by sofroniewn** [comment]
Hi @tibuch so exciting you're willing to help out - there are some underlying issues that need to be sorted out around our layer/ data model - but we're actually in the process of trying to set up a working group that would focus on this area and that you'd be more than welcome to join!

Not sure if you're in the napari zulip https://napari.zulipchat.com/login/ yet, but maybe you join that and i can send you some details for our developer/ community meetings. We've got a couple, so depending on what timezone you're in you can decide which one you'd like to join and we can pick things up from there

**Comment by psobolewskiPhD** [comment]
This issue is essentially a duplicate of: https://github.com/napari/napari/issues/662 so linking them together
